cql3/query_processor.hh

```diff
@@ -412,13 +419,32 @@ private:
                 assert(bound_terms == prepared->bound_names.size());
                 return make_ready_future<std::unique_ptr<statements::prepared_statement>>(std::move(prepared));
             }).then([&key, &id_getter, &client_state] (auto prep_ptr) {
+                auto* statement = prep_ptr->statement.get();
+
+                bytes statement_type_bytes = serialized("");
+                if (auto p = dynamic_cast<cql3::statements::create_keyspace_statement*>(statement))
+                    statement_type_bytes = serialized("CREATE KEYSPACE");
+                else if (auto p = dynamic_cast<cql3::statements::drop_keyspace_statement*>(statement))
+                    statement_type_bytes = serialized("DROP_KEYSPACE");
+                else if (auto p = dynamic_cast<cql3::statements::create_table_statement*>(statement))
+                    statement_type_bytes = serialized("CREATE TABLE");
+                else if (auto p = dynamic_cast<cql3::statements::drop_table_statement*>(statement))
+                    statement_type_bytes = serialized("DROP TABLE");
+                else if (auto p = dynamic_cast<cql3::statements::select_statement*>(statement))
+                    statement_type_bytes = serialized("SELECT");
+                else if (auto p = dynamic_cast<cql3::statements::delete_statement*>(statement))
+                    statement_type_bytes = serialized("DELETE");
+                else if (auto p = dynamic_cast<cql3::statements::update_statement*>(statement))
+                    statement_type_bytes = serialized("UPDATE");
```

Comment:
I know this pattern is already used in other places in Scylla (and this is what I initially recommended), but a better way would be to add a pure virtual method to `cql_statement`:

```cpp
    virtual cql_statement_type get_statement_type() const = 0;
```

that would return a new enum (`cql_statement_type`). That way there is a compile-time guarantee that all statement types will have this implemented (for example, in your if statements you forgot about `ALTER`). A serialization function (`cql_statement_type` to `bytes`) should be near `cql_statement_type` enum class declaration, so that it's easy to modify all its uses in case a new statement type is added.